crates/aggregator/src/lib.rs

```diff
@@ -0,0 +1,324 @@
+//! Aggregator crate
+
+/// Aggregator Config
+pub mod config;
+/// Aggregator error
+pub mod error;
+/// Signed Task Response
+pub mod signed_task_response;
+/// Traits
+pub mod traits;
+
+use alloy::providers::Provider;
+use alloy::providers::{ProviderBuilder, WsConnect};
+use alloy::rpc::types::Filter;
+use alloy::sol_types::SolEvent;
+use eigen_client_avsregistry::reader::AvsRegistryChainReader;
+use eigen_common::get_ws_provider;
+use eigen_logging::get_logger;
+use eigen_services_avsregistry::chaincaller::AvsRegistryServiceChainCaller;
+use eigen_services_blsaggregation::bls_agg::{
+    AggregateReceiver, BlsAggregatorService, ServiceHandle, TaskSignature,
+};
+use eigen_services_operatorsinfo::operatorsinfo_inmemory::OperatorInfoServiceInMemory;
+use futures_util::StreamExt;
+use jsonrpc_core::serde_json;
+use jsonrpc_core::{Error, IoHandler, Params, Value};
+use jsonrpc_http_server::{AccessControlAllowOrigin, DomainsValidation, ServerBuilder};
+use std::{net::SocketAddr, sync::Arc};
+use tokio::sync::Mutex;
+use tracing::info;
+
+pub use config::AggregatorConfig;
+pub use eigen_services_blsaggregation::{
+    bls_agg::TaskMetadata, bls_aggregation_service_response::BlsAggregationServiceResponse,
+};
+pub use error::AggregatorError;
+pub use signed_task_response::SignedTaskResponse;
+pub use traits::{
+    task_processor::{TaskProcessor, TaskProcessorError},
+    task_response::TaskResponse,
+};
+
+/// Aggregator
+#[derive(Debug)]
+pub struct Aggregator<TP> {
+    port_address: String,
+    task_processor: Arc<Mutex<TP>>,
```

Comment:
I think we should move the `Arc<Mutex>` to the `TaskProcessor` as `Send + Sync + 'static` requirements. Let's discuss this before finalizing the interface.